fact_registry/src/lib.cairo

```diff
@@ -0,0 +1,38 @@
+use cairo_verifier::deserialization::stark::StarkProofWithSerde;
+
+mod component;
+
+#[starknet::interface]
+trait IFactRegistry<TContractState> {
+    fn verify_and_register_fact(ref self: TContractState, stark_proof: StarkProofWithSerde);
+    fn is_valid(self: @TContractState, fact: u256) -> bool;
+}
+
+#[starknet::contract]
+mod FactRegistry {
+    use cairo_verifier::{
+        deserialization::stark::StarkProofWithSerde, stark::{StarkProof, StarkProofTrait},
+    };
+    use core::keccak::keccak_u256s_be_inputs;
+
+    #[storage]
+    struct Storage {
+        facts: LegacyMap<u256, bool>,
```

Comment:
As mentioned to @neotheprogramist, how viable is it to change this Boolean to an enumeration to enable optimistic facts registration?

With this pattern, a client that trusts the operator may choose to act before the proof is actually available.

Adding an enumeration with for example `NotFound`, `Registered`, `Proven` without the possibility to go down in the enumeration could allow such behavior.

There are additional security concerns in the code for sure to handle this enumeration which have to be considered, however this may bring an opening for clients depending on the proof generation time.